work/views.py

```diff
@@ -986,7 +995,7 @@ def get_queryset_by_request_data(self, request_data, sort_by=None) -> Iterable:
         queries.extend(
             query_utils.create_queries_by_lookup_field(request_data, self.search_fields, search_fields_maps)
         )
-        return self.create_queryset_by_queries(CofkUnionQueryableWork, queries, sort_by=sort_by).distinct()
+        return self.create_queryset_by_queries(CofkUnionQueryableWork, queries, sort_by=sort_by)
```

Comment:
distinct can be remove in location query is because it use exist subquery. So, if you want to remove distinct you have to use exist subquery

Reply:
Reverted to use distinct